hail/src/main/scala/is/hail/expr/types/physical/PCanonicalCall.scala

```diff
@@ -56,4 +57,18 @@ final case class PCanonicalCall(required: Boolean) extends PCall {
       )
     }
 
+    override def storeShallowAtOffset(dstAddress: Code[Long], valueAddress: Code[Long]): Code[Unit] =
+      this.representation.storeShallowAtOffset(dstAddress, valueAddress)
+
+    override def storeShallowAtOffset(dstAddress: Long, valueAddress: Long) {
+      this.representation.storeShallowAtOffset(dstAddress, valueAddress)
+    }
+
+    override def copyFromType(mb: MethodBuilder, region: Code[Region], srcPType: PType, srcAddress: Code[Long], forceDeep: Boolean): Code[Long] = {
```

Comment:
I think that we can `copyFromType/StoreShallowXX` should be implemented by ComplexType to call the `representation`'s methods. This is basically dead code as existing here, since PCAnonicalCall is not a fundamental type and we only call the copy stuff on funadmental types.